internal/host/transition_test.go

```diff
@@ -5988,3 +5989,153 @@ func validateEqualProgress(p1, p2 *models.HostProgressInfo) {
+			func(_ stateswitch.StateSwitch, _ stateswitch.TransitionArgs) error { return nil },
+		).AnyTimes()
+
+		mockTransitionHandler.EXPECT().PostRefreshLogsProgress(gomock.Any()).Return(
+			func(_ stateswitch.StateSwitch, _ stateswitch.TransitionArgs) error { return nil },
+		).AnyTimes()
+
+		mockTransitionHandler.EXPECT().PostPreparingForInstallationHost(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+		stateMachine = NewHostStateMachine(stateswitch.NewStateMachine(), mockTransitionHandler)
+
+		testState = newTestState(models.HostStatusKnown)
+
+		allValidationIDs := []validationID{
+			IsMediaConnected,
+			IsConnected,
+			HasInventory,
+			IsMachineCidrDefined,
+			BelongsToMachineCidr,
+			HasMinCPUCores,
+			HasMinValidDisks,
+			HasMinMemory,
+			HasCPUCoresForRole,
+			HasMemoryForRole,
+			IsHostnameUnique,
+			IsHostnameValid,
+			IsIgnitionDownloadable,
+			BelongsToMajorityGroup,
+			IsPlatformNetworkSettingsValid,
+			IsNTPSynced,
+			SucessfullOrUnknownContainerImagesAvailability,
+			AreLsoRequirementsSatisfied,
+			AreOdfRequirementsSatisfied,
+			AreCnvRequirementsSatisfied,
+			AreLvmRequirementsSatisfied,
+			SufficientOrUnknownInstallationDiskSpeed,
+			HasSufficientNetworkLatencyRequirementForRole,
+			HasSufficientPacketLossRequirementForRole,
+			HasDefaultRoute,
+			IsAPIDomainNameResolvedCorrectly,
+			IsAPIInternalDomainNameResolvedCorrectly,
+			IsAppsDomainNameResolvedCorrectly,
+			CompatibleWithClusterPlatform,
+			IsDNSWildcardNotConfigured,
+			DiskEncryptionRequirementsSatisfied,
+			NonOverlappingSubnets,
+			VSphereHostUUIDEnabled,
+			CompatibleAgent,
+			NoSkipInstallationDisk,
+			NoSkipMissingDisk,
+			HostValidationIDServiceHasSufficientSpokeKubeAPIAccess,
+		}
+
+		refreshHostArgs = TransitionArgsRefreshHost{
+			ctx:          ctx,
+			eventHandler: nil,
+			conditions: map[string]bool{
+				string(InstallationDiskSpeedCheckSuccessful): false,
+				string(ClusterPreparingForInstallation):      false,
+				string(ClusterPendingUserAction):             false,
+				string(ClusterInstalling):                    false,
+				string(ValidRoleForInstallation):             true,
+				string(StageInWrongBootStages):               false,
+				string(ClusterInError):                       false,
+				string(SuccessfulContainerImageAvailability): false,
+			},
+			validationResults: map[string]ValidationResults{},
+			db:                &gorm.DB{},
+		}
+
+		for _, validationID := range allValidationIDs {
+			validationCategory, err := validationID.category()
+			Expect(err).ToNot(HaveOccurred())
+			refreshHostArgs.validationResults[validationCategory] = append(
+				refreshHostArgs.validationResults[validationCategory], ValidationResult{
+					ID:      validationID,
+					Status:  ValidationSuccess,
+					Message: "Test validation",
+				})
+			refreshHostArgs.conditions[string(validationID)] = true
+		}
+
+		Expect(stateMachine.Run(TransitionTypeRefresh, testState, &refreshHostArgs)).To(Succeed())
```

Comment:
why do you need it in the before each? isn't that part of the test? 
if it's to bring the host into a known state then again it's very specific for your case and not part of the generic refresh test
you can have a context that will define that you want to start from a known state

Reply:
To ensure we've actually set up the validations correctly by running a refresh and checking that we're still in Known